### PR TITLE
Add missing documentation to `bevy_time`

### PR DESCRIPTION
**Pull request by JoJoJet**
# Objective

* Add documentation to undocumented items in `bevy_time`.
* Add a warning for undocumented items.

### CODE REVIEW

**Review by alice-i-cecile — APPROVED** [review]

**Comment by alice-i-cecile** [comment]
Progress towards https://github.com/bevyengine/bevy/issues/3492

**Comment by Selene-Amanita** [comment]
Not introduced by this PR but can be fixed by it, wrong backticks in https://docs.rs/bevy/0.11.0/bevy/time/fixed_timestep/struct.FixedTime.html#method.expend:
```[`Err(FixedUpdateError`)]```

File: crates/bevy_time/src/lib.rs [hunk 1]
```diff
@@ -69,23 +71,27 @@ impl Plugin for TimePlugin {
 /// you may prefer to set the next [`Time`] value manually.
 #[derive(Resource, Default)]
 pub enum TimeUpdateStrategy {
```
Comment (Selene-Amanita) [inline crates/bevy_time/src/lib.rs:73]: Does this affect `FixedTime`? Should it be mentioned here?

From my understanding `FixedTime` is updated from `Time`, maybe mentioning that briefly here and/or in https://docs.rs/bevy_time/0.11.0/bevy_time/fixed_timestep/index.html would be useful?

File: crates/bevy_time/src/lib.rs [hunk 2]
```diff
@@ -69,23 +71,27 @@ impl Plugin for TimePlugin {
...
+    /// [`Time`] will be automatically updated according to the length of the previous frame.
     #[default]
     Automatic,
-    // Update [`Time`] with an exact `Instant` value
+    /// [`Time`] will be updated to the specified [`Instant`] value.
```
Comment (Selene-Amanita) [inline crates/bevy_time/src/lib.rs:77]:
```suggestion
    /// [`Time`] will be updated to the specified [`Instant`] value each frame.
```

(If I understood correctly)

**Review by Selene-Amanita — COMMENTED** [review]
The doc on this page is great: https://docs.rs/bevy_time/0.11.0/bevy_time/fixed_timestep/index.html but [FixedTime](https://docs.rs/bevy_time/0.11.0/bevy_time/fixed_timestep/struct.FixedTime.html) and [run_fixed_update_schedule](https://docs.rs/bevy_time/0.11.0/bevy_time/fixed_timestep/fn.run_fixed_update_schedule.html) don't link back to it. Since [FixedUpdate](https://docs.rs/bevy/0.11.0/bevy/app/struct.FixedUpdate.html) links to `run_fixed_update_schedule`, I think it would be great to link back to the central doc on fixed time, people (me at least) don't always think to check module documentation.

File: crates/bevy_time/src/lib.rs [hunk 3]
```diff
@@ -69,23 +71,27 @@ impl Plugin for TimePlugin {
 /// you may prefer to set the next [`Time`] value manually.
 #[derive(Resource, Default)]
 pub enum TimeUpdateStrategy {
+    /// [`Time`] will be automatically updated according to the length of the previous frame.
```
[the remark this answers is not in the document]
Comment (JoJoJet) [reply crates/bevy_time/src/lib.rs:74]: Hmm, looks like I was wrong about it using the length of the previous frame. It's updated using an `Instant` sent from the render world.

**Comment by JoJoJet** [comment]
Thank you for the review. I believe I have addressed your points.

**Review by Selene-Amanita — CHANGES_REQUESTED** [review]
Besides a small typo looks good to me!

**Review by Selene-Amanita — APPROVED** [review]

**Comment by GuillaumeGomez** [comment]
To help you with missing docs, you can add `#![deny(missing_docs)]`. :)

**Comment by alice-i-cecile** [comment]
@GuillaumeGomez a warn equivalent was added in `lib.rs` :) Like we discussed in the linked issue, `warn` is less intrusive for development, but still blocks CI.

**Comment by GuillaumeGomez** [comment]
Nice! There is also `rustdoc::missing_doc_code_examples` which could be helpful but I think it's nightly only. Bevy docs is really lacking examples. ^^'

**Pull request merged**